azurerm netapp module crash

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.3.9

### AzureRM Provider Version

3.37.0

### Affected Resource(s)/Data Source(s)

azurerm_netapp_volume

### Terraform Configuration Files

```hcl
resource "azurerm_netapp_account" "netapp_account" {
  name                = local.netapp_account
  location            = azurerm_resource_group.netapp.location
  resource_group_name = azurerm_resource_group.netapp.name

  active_directory {
    username            = "x_dssa1"
    password            = "ikZGfU57dI6df5z7(OT67F9p8"
    smb_server_name     = "pav2m1anfs"
    dns_servers         = ["10.97.6.4", "10.97.6.5"]
    domain              = "int.msci.com"
    organizational_unit = "OU=NetAppFiles,OU=Azure,OU=Servers,OU=MSCI"
  }
}

#------------------Gem4 Netapp Creation -------------------------------

resource "azurerm_netapp_pool" "netapp_pool" {
  name                = local.netapp_pool_gem4
  location            = azurerm_resource_group.netapp.location
  resource_group_name = azurerm_resource_group.netapp.name
  account_name        = azurerm_netapp_account.netapp_account.name
  service_level       = "Standard"
  size_in_tb          = 10
}

resource "azurerm_netapp_pool" "netapp_pool_replica" {
  name                = "${local.netapp_pool_gem4}-Mirror"
  location            = azurerm_resource_group.netapp.location
  resource_group_name = azurerm_resource_group.netapp.name
  account_name        = azurerm_netapp_account.netapp_account.name
  service_level       = "Standard"
  size_in_tb          = 10
}


#policy creation


resource "azurerm_netapp_snapshot_policy" "policygem4" {
  name                = "${local.netapp_pool_gem4}-snapshotpolicy"
  location            = azurerm_resource_group.netapp.location
  resource_group_name = azurerm_resource_group.netapp.name
  account_name        = azurerm_netapp_account.netapp_account.name
  enabled             = true


  daily_schedule {
    snapshots_to_keep = 3
    hour              = 20
    minute            = 15
  }

  monthly_schedule {
    snapshots_to_keep = 1
    days_of_month     = [1, 15, 30]
    hour              = 5
    minute            = 45
  }
}


resource "azurerm_netapp_volume" "netapp_volume" {
  name                       = local.netapp_volume_gem4
  location                   = azurerm_resource_group.netapp.location
  zone                       = "1"
  resource_group_name        = azurerm_resource_group.netapp.name
  account_name               = azurerm_netapp_account.netapp_account.name
  pool_name                  = azurerm_netapp_pool.netapp_pool.name
  volume_path                = local.netapp_volume_gem4
  service_level              = "Standard"
  subnet_id                  = azurerm_subnet.azure-netapp-sbnt.id
  network_features           = "Standard"
  protocols                  = ["NFSv3"]
  security_style             = "Unix"
  storage_quota_in_gb        = "15000"
  snapshot_directory_visible = false

    export_policy_rule {
    rule_index        = 1
    allowed_clients   = ["0.0.0.0/0"]
    protocols_enabled = ["NFSv3"]
    unix_read_only    = false
    unix_read_write   = true
  }

data_protection_snapshot_policy {
    snapshot_policy_id = azurerm_netapp_snapshot_policy.policygem4.id
  }

}

resource "azurerm_netapp_volume" "netapp_volume_replica" {

  depends_on = [azurerm_netapp_volume.netapp_volume]
  name                       = "${local.netapp_volume_gem4}-mirror"
  location                   = azurerm_resource_group.netapp.location
  zone                       = "2"
  resource_group_name        = azurerm_resource_group.netapp.name
  account_name               = azurerm_netapp_account.netapp_account.name
  pool_name                  = azurerm_netapp_pool.netapp_pool_replica.name
  volume_path                = "${local.netapp_volume_gem4}-mirror"
  service_level              = "Standard"
  subnet_id                  = azurerm_subnet.azure-netapp-sbnt.id
  network_features           = "Standard"
  protocols                  = ["NFSv3","CIFS"]
  security_style             = "Unix"
  storage_quota_in_gb        = 15000
  snapshot_directory_visible = false

  export_policy_rule {
    rule_index        = 1
    allowed_clients   = ["0.0.0.0/0"]
    protocols_enabled = ["NFSv3"]
    unix_read_only    = false
    unix_read_write   = true
  }

  # to enable Cross-Region Replication feature
  data_protection_replication {
    endpoint_type             = "dst"
    remote_volume_location    = azurerm_resource_group.netapp.location
    remote_volume_resource_id = azurerm_netapp_volume.netapp_volume.id
    replication_frequency     = "hourly"
  }
}
```


### Debug Output/Panic Output

```shell
azurerm_netapp_volume.netapp_volume: Refreshing state... [id=/subscriptions/2d81aa64-a116-49e8-84ff-14d6b2414864/resourceGroups/rgrp-pva2-me-netapp/providers/Microsoft.NetApp/netAppAccounts/netapp-pva2-me-netapp001/capacityPools/Gem4/volumes/gem4]

Error: Plugin did not respond

  with azurerm_netapp_volume.netapp_volume,
  on main.tf line 85, in resource "azurerm_netapp_volume" "netapp_volume":
  85: resource "azurerm_netapp_volume" "netapp_volume" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
details.

Stack trace from the terraform-provider-azurerm_v3.37.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x5332b0c]

goroutine 274 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp.flattenNetAppVolumeDataProtectionSnapshotPolicy(...)
	github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/netapp_volume_resource.go:1164
github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp.resourceNetAppVolumeRead(0xc002143780, {0x5c8f5a0?, 0xc000101000?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/netapp_volume_resource.go:633 +0xaec
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x6ef4800?, {0x6ef4800?, 0xc002687dd0?}, 0xd?, {0x5c8f5a0?, 0xc000101000?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0013d2540, {0x6ef4800, 0xc002687dd0}, 0xc0026f2000, {0x5c8f5a0, 0xc000101000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000f790e0, {0x6ef4800?, 0xc002687cb0?}, 0xc0026e41c0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:613 +0x4a5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00144fcc0, {0x6ef4800?, 0xc0026876e0?}, 0xc002177620)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:748 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x6340f60?, 0xc00144fcc0}, {0x6ef4800, 0xc0026876e0}, 0xc000d81ce0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001e01e0, {0x6f069a0, 0xc00159a1a0}, 0xc00268cfc0, 0xc00158a3c0, 0xb1b6170, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0001e01e0, {0x6f069a0, 0xc00159a1a0}, 0xc00268cfc0, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.37.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

it should create new volume

### Actual Behaviour

plugin crash in azurerm

### Steps to Reproduce

terraform apply thats sitthat

### Important Factoids

Running for East US2

### References

Nope

--- a/internal/services/netapp/netapp_volume_resource.go
+++ b/internal/services/netapp/netapp_volume_resource.go
@@ -1161,7 +1161,7 @@ func flattenNetAppVolumeDataProtectionReplication(input *volumes.VolumePropertie
 }
 
 func flattenNetAppVolumeDataProtectionSnapshotPolicy(input *volumes.VolumePropertiesDataProtection) []interface{} {
-	if input == nil || input.Snapshot == nil || *input.Snapshot.SnapshotPolicyId == "" {
+	if input == nil || input.Snapshot == nil || input.Snapshot.SnapshotPolicyId == nil || *input.Snapshot.SnapshotPolicyId == "" {
 		return []interface{}{}
 	}
 

